tedana/io.py

```diff
@@ -328,6 +328,51 @@ def add_decomp_prefix(comp_num, prefix, max_value):
     return comp_name
 
 
+def denoise_ts(data, mmix, mask, comptable):
+    """Apply component classifications to data for denoising.
+
+    Parameters
+    ----------
+    data : (S x T) array_like
+        Input time series
+    mmix : (C x T) array_like
+        Mixing matrix for converting input data to component space, where `C`
+        is components and `T` is the same as in `data`
+    mask : (S,) array_like
+        Boolean mask array
+    comptable : (C x X) :obj:`pandas.DataFrame`
+        Component metric table. One row for each component, with a column for
+        each metric. Requires at least one column: "classification".
+
+    Returns
+    -------
+    dnts : (S x T) array_like
+        Denoised data (i.e., data with rejected components removed).
+    hikts : (S x T) array_like
+        High-Kappa data (i.e., data composed only of accepted components).
+    lowkts : (S x T) array_like
+        Low-Kappa data (i.e., data composed only of rejected components).
+    """
+    acc = comptable[comptable.classification == 'accepted'].index.values
+    rej = comptable[comptable.classification == 'rejected'].index.values
+
+    # mask and de-mean data
+    mdata = data[mask]
+    dmdata = mdata.T - mdata.T.mean(axis=0)
+
+    # get variance explained by retained components
+    betas = get_coeffs(dmdata.T, mmix, mask=None)
+    varexpl = (1 - ((dmdata.T - betas.dot(mmix.T))**2.).sum() /
+               (dmdata**2.).sum()) * 100
+    LGR.info('Variance explained by decomposition: {:.02f}%'.format(varexpl))
+
+    # create component and de-noised time series and save to files
```

Comment:
Does this actually save to file here? I thought it just created the relevant time matrices.

Reply:
Good catch! It does not. I must have just copied the comment from the original version.

```suggestion
    # create component-based data
```